pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/realtime/segment/SegmentFlushThresholdComputer.java

```diff
@@ -65,6 +67,16 @@ public int computeThreshold(StreamConfig streamConfig, CommittingSegmentDescript
     long optimalSegmentSizeBytesMin = desiredSegmentSizeBytes / 2;
     double optimalSegmentSizeBytesMax = desiredSegmentSizeBytes * 1.5;
 
+    double segmentSizeVariation = streamConfig.getFlushThresholdVarianceFraction();
+    // Valid value of Segment size variation should be between 0 and 0.5
+    if (segmentSizeVariation < 0.0 || segmentSizeVariation >= 0.5) {
```

Comment:
Consider moving this validation inside `StreamConfig.extractFlushThresholdVarianceFraction()`

Reply:
done